File: c++/leetcode/55. Jump Game.cpp
class Solution
{
public:
  bool canJump(vector<int> &nums)
  {
    const int sz = nums.size();
    int last = sz - 1;

    for (int i = last; i >= 0; --i)
    {
      if (i + nums[i] >= last)
        last = i;
    }

    return last == 0;
  }
};

File: javascript/leetcode/55. Jump Game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function (nums) {
  const sz = nums.length;
  let last = sz - 1;

  for (let i = last; i >= 0; --i) {
    if (i + nums[i] >= last) last = i;
  }

  return last == 0;
};
